packages/search-ui/src/SearchDriver.ts

```diff
@@ -277,13 +279,38 @@ class SearchDriver {
       ...searchParameters
     };
 
+    if (initialResponseState) {
+      const { current, resultsPerPage } = this.state;
+
+      const { totalResults } = initialResponseState;
+
+      // Results paging start & end
+      const start = totalResults === 0 ? 0 : (current - 1) * resultsPerPage + 1;
+
+      const end =
+        totalResults < start + resultsPerPage
+          ? totalResults
+          : start + resultsPerPage - 1;
+
+      this.state = {
```

Comment:
Feels a bit odd to be setting the state above and the resetting it here?

Perhaps create the initial state above as a blank, then populate it with this if present and blend it into the state definition above?

Reply:
Done

Reply:
Sort of.